File: lib/data/models/food/food_serving_model.dart
class FoodServingModel {
  const FoodServingModel({
    this.servingId,
    this.servingDescription,
    this.servingUrl,
    this.metricServingAmount,
    this.metricServingUnit,
    this.numberOfUnits,
    this.measurementDescription,
    this.calories,
    this.carbohydrate,
    this.protein,
    this.fat,
    this.saturatedFat,
    this.polyunsaturatedFat,
    this.monounsaturatedFat,
    this.cholesterol,
    this.sodium,
    this.potassium,
    this.fiber,
    this.sugar,
    this.vitaminA,
    this.vitaminC,
    this.calcium,
    this.iron,
  });

  factory FoodServingModel.fromJson(Map<String, dynamic> json) {
    return FoodServingModel(
      servingId: json['serving_id'] as String?,
      servingDescription: json['serving_description'] as String?,
      servingUrl: json['serving_url'] as String?,
      metricServingAmount:
          json['metric_serving_amount'] != null
              ? double.tryParse(json['metric_serving_amount'] as String)
              : null,
      metricServingUnit: json['metric_serving_unit'] as String?,
      numberOfUnits:
          json['number_of_units'] != null
              ? double.tryParse(json['number_of_units'] as String)
              : null,
      measurementDescription: json['measurement_description'] as String?,
      calories:
          json['calories'] != null
              ? int.tryParse(json['calories'] as String)
              : null,
      carbohydrate:
          json['carbohydrate'] != null
              ? double.tryParse(json['carbohydrate'] as String)
              : null,
      protein:
          json['protein'] != null
              ? double.tryParse(json['protein'] as String)
              : null,
      fat: json['fat'] != null ? double.tryParse(json['fat'] as String) : null,
      saturatedFat:
          json['saturated_fat'] != null
              ? double.tryParse(json['saturated_fat'] as String)
              : null,
      polyunsaturatedFat:
          json['polyunsaturated_fat'] != null
              ? double.tryParse(json['polyunsaturated_fat'] as String)
              : null,
      monounsaturatedFat:
          json['monounsaturated_fat'] != null
              ? double.tryParse(json['monounsaturated_fat'] as String)
              : null,
      cholesterol:
          json['cholesterol'] != null
              ? double.tryParse(json['cholesterol'] as String)
              : null,
      sodium:
          json['sodium'] != null
              ? double.tryParse(json['sodium'] as String)
              : null,
      potassium:
          json['potassium'] != null
              ? double.tryParse(json['potassium'] as String)
              : null,
      fiber:
          json['fiber'] != null
              ? double.tryParse(json['fiber'] as String)
              : null,
      sugar:
          json['sugar'] != null
              ? double.tryParse(json['sugar'] as String)
              : null,
      vitaminA:
          json['vitamin_a'] != null
              ? double.tryParse(json['vitamin_a'] as String)
              : null,
      vitaminC:
          json['vitamin_c'] != null
              ? double.tryParse(json['vitamin_c'] as String)
              : null,
      calcium:
          json['calcium'] != null
              ? double.tryParse(json['calcium'] as String)
              : null,
      iron:
          json['iron'] != null ? double.tryParse(json['iron'] as String) : null,
    );
  }

  final String? servingId;
  final String? servingDescription;
  final String? servingUrl;
  final double? metricServingAmount;
  final String? metricServingUnit;
  final double? numberOfUnits;
  final String? measurementDescription;
  final int? calories;
  final double? carbohydrate;
  final double? protein;
  final double? fat;
  final double? saturatedFat;
  final double? polyunsaturatedFat;
  final double? monounsaturatedFat;
  final double? cholesterol;
  final double? sodium;
  final double? potassium;
  final double? fiber;
  final double? sugar;
  final double? vitaminA;
  final double? vitaminC;
  final double? calcium;
  final double? iron;
}

File: server/src/models/foods/FoodServingModel.js
class FoodServingModel {
  constructor({
    id,
    description,
    url,
    metricServingAmount,
    metricServingUnit,
    numberOfUnits,
    measurementDescription,
    calories,
    carbohydrate,
    protein,
    fat,
    saturatedFat,
    polyunsaturatedFat,
    monounsaturatedFat,
    transFat,
    cholesterol,
    sodium,
    potassium,
    fiber,
    sugar,
    addedSugars,
    vitaminA,
    vitaminC,
    vitaminD,
    calcium,
    iron,
  }) {
    this.id = id;
    this.description = description;
    this.url = url;
    this.metricServingAmount = metricServingAmount;
    this.metricServingUnit = metricServingUnit;
    this.numberOfUnits = numberOfUnits;
    this.measurementDescription = measurementDescription;
    this.calories = calories;
    this.carbohydrate = carbohydrate;
    this.protein = protein;
    this.fat = fat;
    this.saturatedFat = saturatedFat;
    this.polyunsaturatedFat = polyunsaturatedFat;
    this.monounsaturatedFat = monounsaturatedFat;
    this.transFat = transFat;
    this.cholesterol = cholesterol;
    this.sodium = sodium;
    this.potassium = potassium;
    this.fiber = fiber;
    this.sugar = sugar;
    this.addedSugars = addedSugars;
    this.vitaminA = vitaminA;
    this.vitaminC = vitaminC;
    this.vitaminD = vitaminD;
    this.calcium = calcium;
    this.iron = iron;
  }

  static fromJson(json) {
    return new FoodServingModel({
      id: json["serving_id"],
      description: json["serving_description"],
      url: json["serving_url"],
      metricServingAmount: json["metric_serving_amount"] ? parseFloat(json["metric_serving_amount"]) : null,
      metricServingUnit: json["metric_serving_unit"],
      numberOfUnits: json["number_of_units"] ? parseFloat(json["number_of_units"]) : null,
      measurementDescription: json["measurement_description"],
      calories: json["calories"] ? parseFloat(json["calories"]) : null,
      carbohydrate: json["carbohydrate"] ? parseFloat(json["carbohydrate"]) : null,
      protein: json["protein"] ? parseFloat(json["protein"]) : null,
      fat: json["fat"] ? parseFloat(json["fat"]) : null,
      saturatedFat: json["saturated_fat"] ? parseFloat(json["saturated_fat"]) : null,
      polyunsaturatedFat: json["polyunsaturated_fat"] ? parseFloat(json["polyunsaturated_fat"]) : null,
      monounsaturatedFat: json["monounsaturated_fat"] ? parseFloat(json["monounsaturated_fat"]) : null,
      transFat: json["trans_fat"] ? parseFloat(json[trans_fat]) : null,
      cholesterol: json["cholesterol"] ? parseFloat(json["cholesterol"]) : null,
      sodium: json["sodium"] ? parseFloat(json["sodium"]) : null,
      potassium: json["potassium"] ? parseFloat(json["potassium"]) : null,
      fiber: json["fiber"] ? parseFloat(json["fiber"]) : null,
      sugar: json["sugar"] ? parseFloat(json["sugar"]) : null,
      addedSugars: json["added_sugars"] ? parseFloat(json["added_sugars"]) : null,
      vitaminA: json["vitamin_a"] ? parseFloat(json["vitamin_a"]) : null,
      vitaminC: json["vitamin_c"] ? parseFloat(json["vitamin_c"]) : null,
      vitaminD: json["vitamin_d"] ? parseFloat(json["vitamin_d"]) : null,
      calcium: json["calcium"] ? parseFloat(json["calcium"]) : null,
      iron: json["iron"] ? parseFloat(json["iron"]) : null,
    });
  }

  toJson() {
    return {
      id: this.id,
      description: this.description,
      url: this.url,
      metricServingAmount: this.metricServingAmount,
      metricServingUnit: this.metricServingUnit,
      numberOfUnits: this.numberOfUnits,
      measurementDescription: this.measurementDescription,
      calories: this.calories,
      carbohydrate: this.carbohydrate,
      protein: this.protein,
      fat: this.fat,
      saturatedFat: this.saturatedFat,
      polyunsaturatedFat: this.polyunsaturatedFat,
      monounsaturatedFat: this.monounsaturatedFat,
      transFat: this.transFat,
      cholesterol: this.cholesterol,
      sodium: this.sodium,
      potassium: this.potassium,
      fiber: this.fiber,
      sugar: this.sugar,
      addedSugars: this.addedSugars,
      vitaminA: this.vitaminA,
      vitaminC: this.vitaminC,
      vitaminD: this.vitaminD,
      calcium: this.calcium,
      iron: this.iron,
    };
  }
}

export default FoodServingModel;
